compiler/rustc_typeck/src/expr_use_visitor.rs

```diff
@@ -28,19 +26,14 @@ use crate::mem_categorization as mc;
 /// This trait defines the callbacks you can expect to receive when
 /// employing the ExprUseVisitor.
 pub trait Delegate<'tcx> {
-    // The value found at `place` is either copied or moved, depending
+    // The value found at `place` is moved, depending
     // on `mode`. Where `diag_expr_id` is the id used for diagnostics for `place`.
     //
     // The parameter `diag_expr_id` indicates the HIR id that ought to be used for
     // diagnostics. Around pattern matching such as `let pat = expr`, the diagnostic
     // id will be the id of the expression `expr` but the place itself will have
     // the id of the binding in the pattern `pat`.
```

Comment:
```suggestion
    // The value found at `place` is moved, meaning that it is used by value
    // and is not known to have `Copy` type. 
    //
    // If a value is used by value but *does* have `Copy` type, that is treated as a
    // "shared borrow". This corresponds to the minimal access that a closure needs
    // to perform that operation (a shared reference).
    //
    // The parameter `diag_expr_id` indicates the HIR id that ought to be used for
    // diagnostics. Around pattern matching such as `let pat = expr`, the diagnostic
    // id will be the id of the expression `expr` but the place itself will have
    // the id of the binding in the pattern `pat`.
```

Reply:
I didn't add the detail about closures because ExprUseVisitor and therefore this delegate is used in other contexts too